### ISSUE TITLE
gate outputs?

### ISSUE THREAD

**Issue by JRMeyer** [jxzhanggg/nonparaSeq2seqVC_code#17]
there are many mentions of `gate` in the code, for example:

https://github.com/jxzhanggg/nonparaSeq2seqVC_code/blob/c977fe9856f4b1d2b494eea4baa320d3e6ecf5c6/fine-tune/model/decoder.py#L241

What is this `gate`, and what does it do?

**Comment by JRMeyer** [comment]
also, here are `gate_targets`:

https://github.com/jxzhanggg/nonparaSeq2seqVC_code/blob/c977fe9856f4b1d2b494eea4baa320d3e6ecf5c6/pre-train/model/loss.py#L32

**Comment by jxzhanggg** [comment]
gate output of decoder is a flag indicating whether the decoding process has reach the last step.
If the decoding has finished, gate output should be >0.5, otherwise, it should be < 0.5.
The second paragraph of Section-III-D of our paper says:
```
In order to end the acoustic feature sequences generated
by the seq2seq decoder at the conversion stage, the hidden
state of the seq2seq decoder at each frame is projected to
a scalar followed by sigmoid activation to predict whether
current frame is the last frame in an utterance. Accordingly,
a cross entropy loss $L_{ED}$ is defined for this prediction at the
training stage.
```

**Comment by JRMeyer** [comment]
I think that `gate` should be changed to another word, because `gate_output` is ambiguous between "the outputs of a gate" and "gate the outputs", and the word `gate` is overloaded in machine learning.

how about `is_final_frame`?

**Comment by jxzhanggg** [comment]
Yes, it's a good suggestion and I think it'll be more clear. 
By the way, the name "gate_output"  actually inherited from the Tacotron 2 repo of NVIDIA.

**Comment by jxzhanggg** [comment]
Because "is_final_frame" is a bit long, I've replace the "gate" with  "stop" in commit f8320cdfe7dccadf118f51bc39be9871485a65b2
